lib/experimental/interactivity-api/navigation-block-interactivity.php

```diff
@@ -0,0 +1,240 @@
+ * >
+ *   <button
+ *     class="wp-block-navigation__responsive-container-open"
+ *     data-wp-on.click="actions.core.navigation.openMenu"
+ *     data-wp-on.keydown="actions.core.navigation.handleMenuKeydown"
+ *   >
+ *   <div
+ *     class="wp-block-navigation__responsive-container"
+ *     data-wp-class.has-modal-open="context.core.navigation.isMenuOpen"
+ *     data-wp-class.is-menu-open="context.core.navigation.isMenuOpen"
+ *     data-wp-bind.aria-hidden="!context.core.navigation.isMenuOpen"
+ *     data-wp-effect="effects.core.navigation.initModal"
+ *     data-wp-on.keydow="actions.core.navigation.handleMenuKeydown"
+ *     data-wp-on.focusout="actions.core.navigation.handleMenuFocusout"
+ *     tabindex="-1"
+ *   >
+ *     <div class="wp-block-navigation__responsive-close">
+ *       <div
+ *         class="wp-block-navigation__responsive-dialog"
+ *         data-wp-bind.aria-modal="context.core.navigation.isMenuOpen"
+ *         data-wp-bind.role="selectors.core.navigation.roleAttribute"
+ *         data-wp-effect="effects.core.navigation.focusFirstElement"
+ *       >
+ *         <button
+ *           class="wp-block-navigation__responsive-container-close"
+ *           data-wp-on.click="actions.core.navigation.closeMenu"
+ *         >
+ *           <svg>
+ *         <button>
+ *         MENU ITEMS
+ *       </div>
+ *     </div>
+ *   </div>
+ * </nav>
+ *
+ * @param string $block_content Markup of the navigation block.
+ *
+ * @return string Navigation block markup with the proper directives
+ */
+function gutenberg_block_core_navigation_add_directives_to_markup( $block_content ) {
+	$w = new WP_HTML_Tag_Processor( $block_content );
+	// Add directives to the `<nav>` element.
+	if ( $w->next_tag( 'nav' ) ) {
+		$w->set_attribute( 'data-wp-island', '' );
+		$w->set_attribute( 'data-wp-context', '{ "core": { "navigation": { "isMenuOpen": false, "overlay": true, "roleAttribute": "" } } }' );
+	};
+
+	// Add directives to the open menu button.
+	if ( $w->next_tag(
+		array(
+			'tag_name'   => 'BUTTON',
+			'class_name' => 'wp-block-navigation__responsive-container-open',
+		)
+	) ) {
+		$w->set_attribute( 'data-wp-on.click', 'actions.core.navigation.openMenu' );
+		$w->set_attribute( 'data-wp-on.keydown', 'actions.core.navigation.handleMenuKeydown' );
+		$w->remove_attribute( 'data-micromodal-trigger' );
+	} else {
+		// If the open modal button not found, we handle submenus immediately.
+		$w = new WP_HTML_Tag_Processor( $w->get_updated_html() );
+
+		// Add directives to the menu container.
+		if ( $w->next_tag(
+			array(
+				'tag_name'   => 'UL',
+				'class_name' => 'wp-block-navigation__container',
+			)
+		) ) {
+			$w->set_attribute( 'data-wp-class.is-menu-open', 'context.core.navigation.isMenuOpen' );
+			$w->set_attribute( 'data-wp-bind.aria-hidden', '!context.core.navigation.isMenuOpen' );
+			$w->set_attribute( 'data-wp-effect', 'effects.core.navigation.initModal' );
```

Comment:
@SantosGuillamot, would it make sense to rename the effect to reflect the fact it works with the menu when there is no modal, too?

Reply:
I renamed it to `initMenu` in [this commit](https://github.com/WordPress/gutenberg/pull/50670/commits/0ce72a818dc3252be365566e00c2b9f8c0556fb0).